### PR TITLE
chore(mep): Remove metrics dry run

### PR DESCRIPTION
**Pull request by wmak**
- We no longer need this dry run since we're now running metrics queries in production

### CODE REVIEW

File: src/sentry/features/__init__.py [hunk 1]
```diff
@@ -112,8 +112,10 @@
 default_manager.add("organizations:org-subdomains", OrganizationFeature)
 default_manager.add("organizations:performance-anomaly-detection-ui", OrganizationFeature, True)
 default_manager.add("organizations:performance-chart-interpolation", OrganizationFeature, True)
-default_manager.add("organizations:performance-dry-run-mep", OrganizationFeature, True)
-default_manager.add("organizations:performance-n-plus-one-api-calls-detector", OrganizationFeature)
+default_manager.add("organizations:performance-frontend-use-events-endpoint", OrganizationFeature, True)
+default_manager.add("organizations:performance-issues", OrganizationFeature, True)
+default_manager.add("organizations:performance-issues-ingest", OrganizationFeature)
+default_manager.add("organizations:performance-issues-post-process-group", OrganizationFeature)
```
Comment (narsaynorath) [inline src/sentry/features/__init__.py:118]: These look like they were added by mistake, I expected to only see the removal of the dry run flag

File: tests/sentry/search/events/builder/test_metrics.py [hunk 2]
```diff
@@ -1452,84 +1451,6 @@ def test_multiple_dataset_but_no_data(self):
...
-                "count_web_vitals(measurements.lcp, good)",
-            ],
-            dry_run=True,
-        )
-        query.run_query("test_query")
-        assert not mock_indexer.called
-        assert not mock_query.called
-
-    @mock.patch("sentry.search.events.builder.metrics.indexer.resolve", return_value=-1)
-    def test_multiple_references_only_resolve_index_once(self, mock_indexer):
```
Comment (narsaynorath) [inline tests/sentry/search/events/builder/test_metrics.py:1477]: Is this still a valid test? It wasn't setting `dry_run=True` in the constructor

File: tests/sentry/search/events/builder/test_metrics.py [hunk 3]
```diff
@@ -1452,84 +1451,6 @@ def test_multiple_dataset_but_no_data(self):
...
-                "sum(measurements.custom.measurement)",
-                "max(measurements.custom.measurement)",
-                "min(measurements.custom.measurement)",
-                "count_unique(user)",
-            ],
-            query="transaction:foo_transaction",
-            allow_metric_aggregates=False,
-            use_aggregate_conditions=True,
-            # Use dry run for now to not hit indexer
-            dry_run=True,
```
Comment (narsaynorath) [inline tests/sentry/search/events/builder/test_metrics.py:1530]: Is this test still valid and instead we should just remove the dry_run arg?

**Review by narsaynorath — APPROVED** [review]

File: src/sentry/features/__init__.py (hunk 1 above)
Comment (wmak) [reply src/sentry/features/__init__.py:118]: They were good catch

File: tests/sentry/search/events/builder/test_metrics.py (hunk 2 above)
Comment (wmak) [reply tests/sentry/search/events/builder/test_metrics.py:1477]: Ah you're right :+1:

**Pull request merged**